src/attributes.md

```diff
@@ -82,6 +81,47 @@ fn some_unused_variables() {
 }
 ```
 
+There are three kinds of attributes:
+
+* Built-in attributes
+* Macro attributes
+* Derive mode helper attributes
+
+## Dynamic and inert attributes
+
+An attribute is either dynamic or inert. During attribute processing, *dynamic
+attributes* remove themselves from the thing they are on while *inert attriutes*
+stay on.
+
+The `cfg` and `cfg_attr` attributes are dynamic. The `test` attribute is inert
+when compiling for tests and dynamic otherwise. Attribute macros are dynamic.
+All other attributes are inert. 
+
+## Attribute resolution
```

Comment:
I'd prefer to avoid specifying any details here for now, the rules are not settled yet.
Macro modularization was stabilized, but there's still a number of incomplete items from https://github.com/rust-lang/rust/pull/50911#issuecomment-411605393 that need to be done until beta, stable or maybe even later if crater allows.
(My suggestion would be to stash this section somewhere until 1.30 stable is released.)